wallet-new/server/Cardano/Wallet/API/V1/Handlers/Wallets.hs

```diff
@@ -57,20 +59,27 @@ listWallets PaginationParams {..} = do
       }
     _ -> return $ OneOf $ Left example
 
-updatePassword :: WalletId
-               -> PasswordUpdate
-               -> MonadV1 Wallet
+updatePassword
+    :: WalletId
+    -> PasswordUpdate
+    -> MonadV1 Wallet
 updatePassword _ _ = liftIO $ generate arbitrary
 
-deleteWallet :: WalletId
-             -> MonadV1 NoContent
-deleteWallet _ = return NoContent
+-- | Deletes an exisiting wallet.
+deleteWallet
+    :: (HasConfigurations, MonadWalletLogic ctx m)
+    => WalletId
+    -> m NoContent
+deleteWallet (WalletId walletId) =
+    V0.deleteWallet . V0.CId . V0.CHash $ walletId
```

Comment:
Just a mental note: We are currently not doing any error handling yet, but obviously in a production-quality setting we would need to check the deletion succeeded etc. But this is for post-phase 1, no need to worry about this right now.

Reply:
Yes, it occurred to me, but I realized that writing something like:
```
case V0.deleteWallet . V0.CId . V0.CHash $ walletId of
    NoContent -> NoContent
    otherwise -> throwM $ WalletError $ "Error with wallet deletion."
```
Is the same it's now, since the type system enforces that `NoContent` is returned and in the other cases `MonadThrow` is used.